client/js/libs/handlebars/ircmessageparser/merge.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+const anyIntersection = require("./anyIntersection");
+const fill = require("./fill");
+
+function Object_assign(target) {
+	Array.prototype.slice.call(arguments, 1).forEach(function(obj) {
+		Object.keys(obj).forEach(function(key) {
+			target[key] = obj[key];
+		});
+	});
+	return target;
+}
+
+function merge(textParts, styleFragments) {
+	const cleanText = styleFragments.map(fragment => fragment.text).join("");
+
+	const allParts = textParts.concat(fill(textParts, cleanText))
+        .sort((a, b) => a.start - b.start);
+
+	return allParts.map(textPart => {
+		textPart.fragments = styleFragments
+            .filter(fragment => anyIntersection(textPart, fragment))
```

Comment:
you got the concept of the parser?
you get a message like `\x02irc\x0f://\x1dfreenode.net\x0f/\x034,8thelounge`
1. step is extract the styling information and get the plain text
2. on the plain text, find channels and urls, fill rest with simple "text" object
3. merge the styling information with the channels / url / text objects
and in the case of lounge
4. generate html strings with result of step 3